### PR TITLE
Bluetooth: controller: Only allow supported feature bits to be set by host

### LINKED ISSUE #54544: Bluetooth: controller: HCI/CCO/BI-45-C setHostFeatureBit failing
**Describe the bug**
The function ll_set_host_feature in ll_feat.c allows any host-controlled feature to be set, not only features that are actually supported.

> Core Spec v5.3, Vol 4, Part E, 7.8.115 LE Set Host Feature command: If Bit_Value is set to 0x01 and Bit_Number specifies a feature bit that requires support of a feature that the Controller does not support, the Controller shall return the error code Unsupported Feature or Parameter Value (0x11)

**Expected behavior**
Only host-controlled feature bits that are supported can bet set and conformance test passes

**Impact**
Bluetooth controller cannot pass Bluetooth conformance tests for the LE Set Host Feature command

### PR DESCRIPTION
**Pull request by wopu-ot**
Redefine mask of host-controlled feature bits to include only features
that are supported by the controller. This fixes a conformance test
failure where setting an unsupported host-controlled feature bit was
not rejected as it should.

Fixes #54544

Signed-off-by: Wolfgang Puffitsch <wopu@demant.com>

### CODE REVIEW

**Review by cvinayak — DISMISSED** [review]

**Comment by wopu-ot** [comment]
@Thalley @cvinayak It seems that some simulation test cases combine a host that is configured for unicast audio with a controller that does not support Connected Isochronous Streams. Should those tests be disabled? Should we delay this change?

**Comment by Thalley** [comment]
> @Thalley @cvinayak It seems that some simulation test cases combine a host that is configured for unicast audio with a controller that does not support Connected Isochronous Streams. Should those tests be disabled? Should we delay this change?

That is a good question. 

It would be a shame to disable those tests (especially since they provide some much needed testing for BAP before the UPF). 

I think we had a similar issue before, where the workaround was that we simply enable the Kconfigs in the test to enable ISO support: https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/bluetooth/bsim_bt/bsim_test_audio/Kconfig

But that isn't enough for this case?

**Comment by wopu-ot** [comment]
Just pretending that certain Kconfigs are set does not work here, because the actual behavior of the controller is affected by the Kconfigs. I agree that it would be a shame to disable the test cases, but the only options I see to keep them enabled and getting in this PR would be some kind of dirty hack. I suggest to hold back this PR until the Nordic controller supports being a CIS peripheral and live with the nonconformance in the meantime.

**Comment by Thalley** [comment]
> Just pretending that certain Kconfigs are set does not work here, because the actual behavior of the controller is affected by the Kconfigs. I agree that it would be a shame to disable the test cases, but the only options I see to keep them enabled and getting in this PR would be some kind of dirty hack. I suggest to hold back this PR until the Nordic controller supports being a CIS peripheral and live with the nonconformance in the meantime.

That works for me :) @cvinayak please let us know if you have any input here

**Comment by cvinayak** [comment]
> I suggest to hold back this PR until the Nordic controller supports being a CIS peripheral and live with the nonconformance in the meantime.

No objections. Fixes will be done as needed when conformance testing is done with support in upstream Zephyr Controller gets in place.

Will you remove the hci_core.c change, so that we can merge the ll_feat.h changes?

**Comment by wopu-ot** [comment]
The changes in ll_feat.h (the redefinition of `LL_FEAT_HOST_BIT_MASK` to be precise) make the test cases fail, there is no point in merging them separately.

**Comment by Thalley** [comment]
@wopu-ot What was the plan for this PR? 

If we are not planning on merging it just yet, can you please either add the DNM label or mark it as draft?

**Comment by wopu-ot** [comment]
Converted to draft until the Nordic controller supports CIS.

**Review by jori-nordic — DISMISSED** [review]

**Comment by Thalley** [comment]
@wopu-ot @cvinayak I guess we can add this now?

**Review by cvinayak — APPROVED** [review]

File: subsys/bluetooth/controller/include/ll_feat.h [hunk 1]
```diff
@@ -276,3 +273,17 @@
 				  LL_FEAT_BIT_ISO_BROADCASTER | \
 				  LL_FEAT_BIT_SYNC_RECEIVER | \
 				  LL_FEAT_BIT_PERIODIC_ADI_SUPPORT)
+
+/* Connected Isochronous Stream (Host Support) bit is controlled by host */
+#if defined(CONFIG_BT_CTLR_CONN_ISO)
+#define LL_FEAT_HOST_BITS_ISO_CHANNELS BIT64(BT_LE_FEAT_BIT_ISO_CHANNELS)
```
Comment (Thalley) [inline subsys/bluetooth/controller/include/ll_feat.h:279]: The naming of `ISO_CHANNELS` in this context is actually pretty confusing, as it is `CIS Host Support`, and doesn't relate to broadcast isochronous channels. We should rename it, but that can done in a separate PR

**Review by Thalley — APPROVED** [review]

**Pull request merged**